x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -745,75 +936,14 @@ func (suite *KeeperTestSuite) TestCalcJoinPoolShares() {
 				sdk.NewInt64Coin("uosmo", 25_000),
 				sdk.NewInt64Coin("uatom", 50_000),
 			),
-			expectShares: sdk.NewInt(1250000000000 + 609374990000),
-		},
-		{
-			// This test doubles the liquidity in a fresh pool, so it should generate the base number of LP shares for pool creation as new shares
-			// This is set to 1e20 (or 100 * 10^18) for Osmosis, so we should expect:
-			// P_issued = 1e20
-			name:    "minimum input with two assets and minimum liquidity",
-			swapFee: sdk.MustNewDecFromStr("0"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 1),
-					Weight: sdk.NewInt(100),
-				},
-				{
-					Token:  sdk.NewInt64Coin("uatom", 1),
-					Weight: sdk.NewInt(100),
-				},
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("uosmo", 1),
-				sdk.NewInt64Coin("uatom", 1),
-			),
-			expectShares: sdk.NewInt(1e18).Mul(sdk.NewInt(100)),
-		},
-		{
-			// Pool liquidity is changed by 1e-12
-			// P_issued = 1e20 * 1e-12 = 1e8
-			name:    "minimum input two assets equal liquidity",
-			swapFee: sdk.MustNewDecFromStr("0"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-				{
-					Token:  sdk.NewInt64Coin("uatom", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("uosmo", 1),
-				sdk.NewInt64Coin("uatom", 1),
-			),
-			expectShares: sdk.NewInt(100_000_000),
-		},
-		{
-			// Input assets do not exist in pool
-			// P_issued = 1e20 * 1e-12 = 1e8
-			name:    "attempt join with denoms not in pool",
-			swapFee: sdk.MustNewDecFromStr("0"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-				{
-					Token:  sdk.NewInt64Coin("uatom", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("foo", 1),
-				sdk.NewInt64Coin("baz", 1),
+			expectShares: sdk.NewInt(1250000000000),
+			expectLiq: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 25_000),
+				sdk.NewInt64Coin("uatom", 12_500),
```

Comment:
From my understanding, the ratio of tokens in relative to the pool amount is what matters for the core of the calculation in `MaximalExactRatioJoin` while the weights are irrelevant.

I know you've mentioned the "ratios" part already in the comment but can we elaborate on this please and show the calculation steps? I think it would be useful to show that the weights have no effect while the ratios do. It would help ease the onboarding on the tests.

Based on my understanding of the calculation:
```
initialShareAmount = 1 * 10**18 * 100

expectShares = 25000 / 2_000_000_000_000 * initialShareAmount = 1250000000000.0

uosmo expected liq = 25_000 because it has the minimal requested token in / current supply ration

uatom = minRatio  * current uatom supply = 25000 / 2_000_000_000_000 * 1_000_000_000_000 = 12_500
```

Reply:
Great point – please see my recent change. I added brief comments explaining the dynamic (ratio matters but weights don't) and how the exact results were obtained